fairlearn/metrics/_metric_frame.py

```diff
@@ -0,0 +1,552 @@
+# Copyright (c) Microsoft Corporation and Fairlearn contributors.
+# Licensed under the MIT License.
+
+import copy
+import logging
+import numpy as np
+import pandas as pd
+from typing import Any, Callable, Dict, List, Optional, Union
```

Comment:
`Callable`, `Dict` and `List` are going to be [deprecated](https://docs.python.org/3/library/typing.html#module-contents). Okay to keep for now, but maybe pro-actively create an issue?

Reply:
I've fixed `Dict` and `List`... will create issue for Callable, since that could get interesting.

Reply:
Will create the issue once this is in. Fixing isn't as easy as I'd hoped.